conda_index/utils_build.py

```diff
@@ -174,7 +174,21 @@ def merge_or_update_dict(
     if base == new:
         return base
 
-    for key, value in new.items():
+    if not add_missing_keys:
```

Comment:
Another part of the effort to make repodata patching more efficient for shards; not entirely successful.

Reply:
@dholth you are not currently storing the patches in the sqlite database, correct? Maybe that could help with the efficiency?

Reply:
We tend to execute code to generate patches every time.

Reply:
Generate patches as in JLAP or apply patches as in repodata patchesd. I understood this code was doing the latter, so I might have misunderstood :)

Reply:
conda-forge downloads repodata, generates hotfixes and then stores those patches in a json file, but in general conda-index executes code to generate hotfixes (unfortunately).